### DISCUSSION TITLE
Combine multiple choice and text box question

### DISCUSSION THREAD

**Discussion by trygveso** [jspsych/jsPsych#3026] [Q&A] [answered]
Hi, 
I need a standard multiple choice question (survey-multi-choice) in which the last option is "other". 
If the user chooses "other", a text box should appear (survey-text). 

Alternatively, it would be ok if the text box was always visible as the last option in a multiple choice question.

I.e.: 
[ ] "green"
[ ] "blue"
[ ] "other: [ 'textbox response' ] "

How can I combine survey-multi-choice and survey-text in a single question?

It seems like the SurveyJS-plugin is no longer supported (?)

Thanks for helping!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @trygveso, I think the SurveyJS plugin is still supported. It's listed in the packages as plugin-survey. The [example file](https://github.com/jspsych/jsPsych/blob/main/packages/plugin-survey/example.html) shows how to include an "other" option with a text box:
  https://github.com/jspsych/jsPsych/blob/74f950b75194d3ed93e4a1af1ff41529d38ff88d/packages/plugin-survey/example.html#L68

Is that what you need?

**Comment by woutvos** [comment]
Hi there, trying to use this function but can't get it working. Does anyone know what I do wrong? Thanks in advance!

```
timeline.push({
  type: jsPsychSurveyMultiChoice,
  questions: [
    {
      prompt: "Some good question?", 
      options: ["option 1", "option 2", "option 3"],
      add_other_option: true,
      horizontal: false,
      required: true,
      name: 'the_good_question'
    }],
  button_label: "Next",
})
```

**Reply by SHogenboom** [reply]
@woutvos ; I think you need to change the `type` and the parameters slightly (based on the linked example). Can you give this a try? Otherwise we’ll wait for the experts to help!

```
timeline.push({
 type: jsPsychSurvey,
     pages: [ {type: 'multi-choice', prompt: 'Some good question? with an "other" option.', 
                     options: ["option 1", "option 2", "option 3”],
                     columns: 1,
                     add_other_option: true}],
      button_label_next: "Next >",
      button_label_back: "< Back",
      button_label_finish: "Finish!",
      show_question_numbers: ‘on’}
```

**Reply by woutvos** [reply]
This worked, thanks!
